site/src/components/ui/multiple-selector.tsx

```diff
@@ -0,0 +1,649 @@
+										onClick={() => handleUnselect(option)}
+									>
+										<X className="h-4 w-4 text-content-secondary hover:text-content-primary align-text-bottom" />
+									</button>
+								</Badge>
+							);
+						})}
+						{/* Avoid having the "Search" Icon */}
+						<CommandPrimitive.Input
+							{...inputProps}
+							ref={inputRef}
+							value={inputValue}
+							disabled={disabled}
+							onValueChange={(value) => {
+								setInputValue(value);
+								inputProps?.onValueChange?.(value);
+							}}
+							onBlur={(event) => {
+								if (!onScrollbar) {
+									setOpen(false);
+								}
+								inputProps?.onBlur?.(event);
+							}}
+							onFocus={(event) => {
+								setOpen(true);
+								triggerSearchOnFocus && onSearch?.(debouncedSearchTerm);
+								inputProps?.onFocus?.(event);
+							}}
+							placeholder={
+								hidePlaceholderWhenSelected && selected.length !== 0
+									? ""
+									: placeholder
+							}
+							className={cn(
+								"flex-1 border-none outline-none bg-transparent placeholder:text-content-secondary",
+								{
+									"w-full": hidePlaceholderWhenSelected,
+									"px-3 py-2.5": selected.length === 0,
+									"ml-1": selected.length !== 0,
+								},
+								inputProps?.className,
+							)}
+						/>
+						<button
+							type="button"
+							onClick={() => {
+								setSelected(selected.filter((s) => s.fixed));
+								onChange?.(selected.filter((s) => s.fixed));
+							}}
+							className={cn(
+								"absolute right-6 mt-1 bg-transparent border-none cursor-pointer text-content-secondary",
+								(hideClearAllButton ||
+									disabled ||
+									selected.length < 1 ||
+									selected.filter((s) => s.fixed).length === selected.length) &&
+									"hidden",
+							)}
+						>
+							<X size={18} />
+						</button>
+						<ChevronDown className="h-6 w-6 cursor-pointer text-content-secondary" />
+					</div>
+				</div>
+				<div className="relative">
+					{open && (
+						<CommandList
+							className="absolute top-1 z-10 w-full rounded-md border border-solid border-border bg-popover text-popover-foreground shadow-md outline-none animate-in"
+							onMouseLeave={() => {
+								setOnScrollbar(false);
+							}}
+							onMouseEnter={() => {
+								setOnScrollbar(true);
+							}}
+							onMouseUp={() => {
+								inputRef?.current?.focus();
+							}}
+						>
+							{isLoading ? (
+								<>{loadingIndicator}</>
+							) : (
+								<>
+									{EmptyItem()}
+									{CreatableItem()}
+									{!selectFirstItem && (
+										<CommandItem value="-" className="hidden" />
+									)}
+									{Object.entries(selectables).map(([key, dropdowns]) => (
+										<CommandGroup
+											key={key}
+											heading={key}
+											className="h-full overflow-auto"
+										>
+											{/* biome-ignore lint/complexity/noUselessFragments: A parent element is
+											    needed for multiple dropdown items */}
```

Comment:
Could you explain this comment a bit more? Is this a library need? And just making sure: is it okay that the element used is never rendered on screen?